Assets/PatchKit Patcher/Scripts/Debug/DebugMenu.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Diagnostics;
+using PatchKit.Unity.Patcher;
+using PatchKit.Unity.Patcher.Debug;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace PatchKit_Patcher.Scripts.Debug
+{
+    public class DebugMenu : MonoBehaviour
+    {
+        private static readonly DebugLogger DebugLogger = new DebugLogger(typeof(DebugMenu));
+
+        private Rect _popUp, _debugMenu;
+        private bool _show;
+        private bool _showPopUp;
+        private string _message;
+        private GraphicRaycaster _graphicRaycaster;
+
+        void Start()
+        {
+            int windowWidth = 250;
+            int windowHeight = 180;
+            int x = (Screen.width - windowWidth) / 2;
+            int y = (Screen.height - windowHeight) / 2;
+            int yPopUp = (Screen.height - 120) / 2;
+            _debugMenu = new Rect(x, y, windowWidth, windowHeight);
+            _popUp = new Rect(x, yPopUp, windowWidth, 120);
+            _graphicRaycaster = FindObjectOfType<GraphicRaycaster>();
+        }
+
+        void OnGUI()
+        {
+            if (Event.current.Equals(Event.KeyboardEvent(KeyCode.D.ToString())))
+            {
+                _show = !_show;
+                _graphicRaycaster.enabled = !_graphicRaycaster.enabled;
```

Comment:
Please introduce a new method `Open` that would open the debug menu.

Later on, you can introduce here an if statement, and depending on if `_show` is `true` or `false`, you can call `Open` or `Close`.